### PR TITLE
Add `CodeSigningConfig` resource

### PR DESCRIPTION
**Pull request by A-Hilaly**
Issue aws-controllers-k8s/community#238

Description of changes:
- Adds create, delete and update operations for `CodeSigningConfig`
resource in the AWS Lambda API.
- Adds end2end tests to deploy lambda code signing configurations
using ACK controller.

By submitting this pull request, I confirm that my contribution is made
under the terms of the Apache 2.0 license.

Issue #, if available:

Description of changes:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by A-Hilaly** [comment]
This resource was generated by making small changes to the code-generator.
/hold for now

File: test/e2e/tests/test_code_signing_config.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        assert cr is not None
+        assert k8s.get_resource_exists(ref)
+
+        codeSigningConfigARN = cr['status']['ackResourceMetadata']['arn']
+
+        time.sleep(CREATE_WAIT_AFTER_SECONDS)
+
+        # Check Lambda code singing config exists
```
Comment (jaypipes) [inline test/e2e/tests/test_code_signing_config.py:90]: s/singing/signing/ (though I like the idea of a singing config ;) )

File: test/e2e/tests/test_code_signing_config.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        assert csc is not None
+        assert csc["Description"] == "new description"
+
+        # Delete k8s resource
+        _, deleted = k8s.delete_custom_resource(ref)
+        assert deleted
+
+        time.sleep(DELETE_WAIT_AFTER_SECONDS)
+
+        # Check Lambda code singing config doesn't exists
```
Comment (jaypipes) [inline test/e2e/tests/test_code_signing_config.py:112]: s/singing/signing/
s/doesn't exists/doesn't exist/

**Review by jaypipes — CHANGES_REQUESTED** [review]
teeny little typo nits inline, otherwise this looks great, thanks @A-Hilaly!

File: test/e2e/tests/test_code_signing_config.py (hunk 1 above)
Comment (A-Hilaly) [reply test/e2e/tests/test_code_signing_config.py]: hahaha :facepalm:

**Review by jaypipes — APPROVED** [review]
:+1: singing in the rain :musical_note:

**Pull request merged**